geoh5py/shared/entity_type.py

```diff
@@ -20,136 +20,209 @@
-        self.workspace.update_attribute(self, "attributes")
+
+        if hasattr(self, "_on_file"):
+            self.workspace.update_attribute(self, "attributes")
 
     @classmethod
     def find(
-        cls: type[EntityTypeT], workspace: ws.Workspace, type_uid: uuid.UUID
+        cls: type[EntityTypeT], workspace: Workspace, type_uid: uuid.UUID
     ) -> EntityTypeT | None:
-        """Finds in the given Workspace the EntityType with the given UUID for
+        """
+        Finds in the given Workspace the EntityType with the given UUID for
         this specific EntityType implementation class.
 
         :return: EntityType of None
         """
         return cast(EntityTypeT, workspace.find_type(type_uid, cls))
 
-    @property
-    def on_file(self) -> bool:
-        """
-        :obj:`bool` Entity already present in
-        :obj:`~geoh5py.workspace.workspace.Workspace.h5file`.
+    @classmethod
+    @abstractmethod
+    def find_or_create(cls, workspace: Workspace, **kwargs) -> EntityType:
         """
-        return self._on_file
+        Find or creates an EntityType with given UUID that matches the given
+        Entity implementation class.
 
-    @on_file.setter
-    def on_file(self, value: bool):
-        self._on_file = value
+        :param workspace: An active Workspace class
 
-    @staticmethod
-    @abstractmethod
-    def _is_abstract() -> bool:
-        """Trick to prevent from instantiating abstract base class."""
-        return True
+        :return: EntityType
+        """
 
     @property
     def name(self) -> str | None:
+        """
+        The name of the entity type.
+        """
         return self._name
 
     @name.setter
-    def name(self, name: str):
+    def name(self, name: str | None):
+        if not isinstance(name, (str, type(None))):
+            raise TypeError(f"name must be a string or None, not {type(name)}")
+
         self._name = name
-        self.workspace.update_attribute(self, "attributes")
+
+        if hasattr(self, "_on_file"):
+            self.workspace.update_attribute(self, "attributes")
+
+    @property
+    def on_file(self) -> bool:
+        """
+        Return True if Entity already present in
+        the workspace.
+        """
+        return self._on_file
+
+    @on_file.setter
+    def on_file(self, value: bool):
+        if not isinstance(value, bool) and value != 1 and value != 0:
+            raise TypeError(f"on_file must be a bool, not {type(value)}")
+        self._on_file = bool(value)
 
     @property
     def uid(self) -> uuid.UUID:
         """
-        :obj:`uuid.UUID` The unique identifier of an entity, either as stored
+        The unique identifier of an entity, either as stored
         in geoh5 or generated in :func:`~uuid.UUID.uuid4` format.
         """
         return self._uid
 
     @uid.setter
-    def uid(self, uid: str | uuid.UUID):
+    def uid(self, uid: str | uuid.UUID | None):
+        if uid is None:
+            uid = uuid.uuid4()
```

Comment:
```suggestion
    def uid(self, uid: str | uuid.UUID):
```

Should never be None

Reply:
Should we have a uid setter?
I don't think the uid of an object should change in a process no?

Reply:
use utility function to set uid type